plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -263,330 +132,156 @@ def sequence_files(
+        if node:
+            return node
+        skip_reason = self._find_skip_reason(project, expected_node_path)
+        if skip_reason:
+            raise SQLFluffSkipFile(f"Skipped file {fname} because it is {skip_reason}")
+        raise SQLFluffSkipFile(
+            f"File {fname} was not found in dbt project"
+        )  # pragma: no cover
 
     @large_file_check
-    def process(self, *, fname, in_str=None, config=None, formatter=None):
-        """Compile a dbt model and return the compiled SQL.
-
-        Args:
-            fname (:obj:`str`): Path to dbt model(s)
-            in_str (:obj:`str`, optional): This is ignored for dbt
-            config (:obj:`FluffConfig`, optional): A specific config to use for this
-                templating operation. Only necessary for some templaters.
-            formatter (:obj:`CallbackFormatter`): Optional object for output.
-        """
+    def process(
+        self,
+        *,
+        in_str: str,
+        fname: Optional[str] = None,
+        config: Optional[FluffConfig] = None,
+        formatter=None,
+        **kwargs,
+    ):
+        """Compile a dbt model and return the compiled SQL."""
         # Stash the formatter if provided to use in cached methods.
         self.formatter = formatter
         self.sqlfluff_config = config
-        self.project_dir = self._get_project_dir()
-        self.profiles_dir = self._get_profiles_dir()
-        fname_absolute_path = os.path.abspath(fname)
-
         try:
-            os.chdir(self.project_dir)
-            processed_result = self._unsafe_process(fname_absolute_path, in_str, config)
-            # Reset the fail counter
-            self._sequential_fails = 0
-            return processed_result
+            return self._unsafe_process(
+                os.path.abspath(fname) if fname else None, in_str, config
+            )
         except DbtCompilationException as e:
-            # Increment the counter
-            self._sequential_fails += 1
             if e.node:
                 return None, [
                     SQLTemplaterError(
                         f"dbt compilation error on file '{e.node.original_file_path}', "
-                        f"{e.msg}",
-                        # It's fatal if we're over the limit
-                        fatal=self._sequential_fails > self.sequential_fail_limit,
+                        f"{e.msg}"
                     )
                 ]
             else:
                 raise  # pragma: no cover
-        except DbtFailedToConnectException as e:
-            return None, [
-                SQLTemplaterError(
-                    "dbt tried to connect to the database and failed: you could use "
-                    "'execute' to skip the database calls. See"
-                    "https://docs.getdbt.com/reference/dbt-jinja-functions/execute/ "
-                    f"Error: {e.msg}",
-                    fatal=True,
-                )
-            ]
```

Comment:
If we remove this catch - how does this now fail in the case of an unreachable DB?

Reply:
Good point. I definitely need to test this, at least manually. I'll add a checkbox item to the PR so I don't forget.

Reply:
Tested, and it crashes with a runtime error. I'll add it back.